buildtest/utils/file.py

```diff
@@ -143,3 +143,66 @@ def resolve_path(path):
 
     if os.path.exists(real_path):
         return real_path
+
+
+def read_file(filepath):
+    """ This method provides capability to read a file.
+
+        Parameters:
+
+        :param filepath: file name to read
+        :type filepath: str, required
+        :return: return content of file
+        :rtype: list
+    """
+
+    # type check filepath to ensure its a string, if not return None
+    if not isinstance(filepath, str):
+        return None
+
+    # resolve_path will handle shell and user expansion and account for any symlinks and check for file existence.
+    # if resolve_path does not return gracefully it implies file does not exist and will return None
+    filepath = resolve_path(filepath)
+
+    # if it's invalid file let's return None
+    if not filepath:
+        return None
+
+    with open(filepath, "r") as fd:
+        content = fd.read()
+
+    return content
+
+
+def write_file(filepath, content):
+    """ This method provides capability to write a file.
+
+        Parameters:
+
+        :param filepath: file name to write
+        :type filepath: str, required
+        :param content: content to write to file
+        :type content: str, required
+        :return: return content of file
+        :rtype: list
+    """
+
+    # type check filepath to ensure its a string, if not return None
```

Comment:
You can combine these first two conditions into one if --> return statement.

Reply:
I am not following, you want me to change the comment or are you referring to the two conditions

```
    # type check filepath to ensure its a string, if not return None
    if not isinstance(filepath, str):
        return None

    # ensure content is of type string
    if not isinstance(content, str):
        return None
```

Reply:
```python
if not isinstance(filepath, str) and not isinstance(content, str):
    return None
```

Reply:
I changed the behavior with meaningful exception messages, i kept them separate so i can get coverage for each if condition.